xmlserializer/XmlDocSource.cpp

```diff
@@ -216,6 +225,7 @@ bool CXmlDocSource::isInstanceDocumentValid()
     xmlSchemaFree(pSchema);
     xmlSchemaFreeParserCtxt(pParserCtxt);
     xmlFreeDoc(pSchemaDoc);
+    xmlFree(schemaURI);
```

Comment:
unique_ptr can be provided with a custom deleter (default is delete, not 
free).

On 09/23/2015 05:34 PM, miguelgaio wrote:
>
> In xmlserializer/XmlDocSource.cpp 
> <https://github.com/01org/parameter-framework/pull/248#discussion_r40218494>:
>
> > @@ -216,6 +225,7 @@ bool CXmlDocSource::isInstanceDocumentValid()
> >      xmlSchemaFree(pSchema);
> >      xmlSchemaFreeParserCtxt(pParserCtxt);
> >      xmlFreeDoc(pSchemaDoc);
> > +    xmlFree(schemaURI);
>
> I'm not sure, xmlFree is not delete or free()
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/01org/parameter-framework/pull/248/files#r40218494>.
>

---------------------------------------------------------------------
Intel Corporation SAS (French simplified joint stock company)
Registered headquarters: "Les Montalets"- 2, rue de Paris, 
92196 Meudon Cedex, France
Registration Number:  302 456 199 R.C.S. NANTERRE
Capital: 4,572,000 Euros

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.